app/scripts/background.js

```diff
@@ -367,17 +361,38 @@ function maybeDetectPhishing(theController) {
   );
 }
 
+function stringifyError(error) {
+  return JSON.stringify({
+    message: error.message,
+    name: error.name,
+    stack: error.stack,
+  });
+}
+
 // These are set after initialization
 let connectRemote;
 let connectExternalExtension;
 let connectExternalCaip;
 
 browser.runtime.onConnect.addListener(async (...args) => {
   // Queue up connection attempts here, waiting until after initialization
-  await isInitialized;
+  try {
+    await isInitialized;
```

Comment:
Where the magic happens (at least in the background)!

If we failed to initialize this `await` will throw, so we'll head down to the `catch`. The catch will post a message to the UI telling it to render the error, which in the case of vault corruption, is our vault corruption error.